agr_literature_service/api/crud/topic_entity_tag_crud.py

```diff
@@ -586,6 +587,46 @@ def show_all_reference_tags(db: Session, curie_or_reference_id, page: int = 1,
         return [get_tet_with_names(db, tag, curie_to_name) for tag in all_tet]
 
 
+def get_reference_tags(db: Session, mod_abbreviation: str, days_updated: int = 7):
+
+    current_date = datetime.now()
+    past_date = current_date - timedelta(days=int(days_updated))
+    last_date_updated = past_date.strftime("%Y-%m-%d")
+
+    rows = db.execute(f"SELECT cr.curie, tet.*, u.email "
+                      f"FROM cross_reference cr "
+                      f"JOIN topic_entity_tag tet ON cr.reference_id = tet.reference_id AND cr.curie_prefix = '{mod_abbreviation}' "
+                      f"JOIN topic_entity_tag_source tets ON tet.topic_entity_tag_source_id = tets.topic_entity_tag_source_id "
+                      f"JOIN users u ON tet.updated_by = u.id "
+                      f"WHERE tets.data_provider = '{mod_abbreviation}' "
+                      f"AND tet.date_updated >= '{last_date_updated}'").fetchall()
+
+    tags = [dict(row) for row in rows]
+
+    curie_to_name_mapping = get_curie_to_name_mapping_for_mod(db, mod_abbreviation, last_date_updated)
+
+    data = [get_tet_with_names(db, tag, curie_to_name_mapping) for tag in tags]
+
+    src_rows = db.execute(f"SELECT * FROM topic_entity_tag_source WHERE data_provider = '{mod_abbreviation}'").fetchall()
+    metadata = [dict(row) for row in src_rows]
+
+    return {"metadata": metadata, "data": data}
+
+
+def get_curie_to_name_mapping_for_mod(db, mod_abbreviation, last_date_updated=None):
```

Comment:
maybe we could merge this function and `get_curie_to_name_from_all_tets` and do it for all references if a reference_id is not passed?

Reply:
Did you mean that we can update the "get_curie_to_name_from_all_tets" function to take optional parameters "mod_abbreviation", "reference_curie_or_id", "date_last_updated". If none of these are provided, getting mapping for all papers in tet table? We can do this, but I will just leave as is since the "get_curie_to_name_from_all_tets" function is pretty complicated now.  What do you think?